centreon/packages/ui/src/RichTextEditor/plugins/FloatingLinkEditorPlugin.tsx

```diff
@@ -0,0 +1,245 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext';
+import {
+  $getSelection,
+  $isRangeSelection,
+  $isTextNode,
+  LexicalEditor
+} from 'lexical';
+import { mergeRegister } from '@lexical/utils';
+import { useAtom, useAtomValue, useSetAtom } from 'jotai';
+import { $isLinkNode, TOGGLE_LINK_COMMAND } from '@lexical/link';
+import { useTranslation } from 'react-i18next';
+import { equals, isNil } from 'ramda';
+
+import { Popper, IconButton, Paper, Link, Box } from '@mui/material';
+import EditIcon from '@mui/icons-material/Edit';
+
+import { getSelectedNode } from '../utils/getSelectedNode';
+import { getDOMRangeRect } from '../utils/getDOMRangeRect';
+import { editLinkModeAtom, isInsertingLinkAtom, linkValueAtom } from '../atoms';
+import InputField from '../../InputField/Text';
+import {
+  labelInputLink,
+  labelSavedLink,
+  labelEditLink
+} from '../translatedLabels';
+
+interface UseFloatingLinkEditorProps {
+  editor: LexicalEditor;
+}
+
+interface FloatingLinkEditorProps {
+  editor: LexicalEditor;
+}
+
+interface TooltipPosition {
+  x: number;
+  y: number;
+}
+
+const FloatingLinkEditor = ({
+  editor
+}: FloatingLinkEditorProps): JSX.Element | null => {
+  const nativeSelection = window.getSelection();
+  const rootElement = editor.getRootElement();
+  const { t } = useTranslation();
+  const [tooltipPosition, setTooltipPosition] = useState<TooltipPosition>({
+    x: 0,
+    y: 0
+  });
+
+  const [editMode, setEditMode] = useAtom(editLinkModeAtom);
+  const [linkUrl, setLinkUrl] = useAtom(linkValueAtom);
+
+  const rangeRect = getDOMRangeRect(nativeSelection, rootElement);
+
+  const acceptOrCancelNewLinkValue = useCallback(
+    (event): void => {
+      const { value } = event.target;
+      if (event.key === 'Enter') {
+        event.preventDefault();
+
+        if (value !== '') {
+          editor.dispatchCommand(TOGGLE_LINK_COMMAND, {
+            target: '_blank',
+            url: value
+          });
+        }
+        setEditMode(false);
+      } else if (event.key === 'Escape') {
+        event.preventDefault();
+        setEditMode(false);
+      }
+    },
+    [setEditMode, setLinkUrl]
+  );
+
+  useEffect(() => {
+    if (isNil(rangeRect)) {
+      return;
+    }
+
+    const isNotPositioned = equals(rangeRect.x, 0) && equals(rangeRect.y, 0);
```

Comment:
Can we declare the other boolean value around? (isPositioned): this will make things more coherent and easier to reason about (then call !isNotPisioned or not(isPositioned)